### PR TITLE
xds: remove deprecated usages of xDS

### LINKED ISSUE #8425: Fix deprecated envoy name warnings
Since https://github.com/hashicorp/consul/pull/8424 changed the supported min envoy version to `v1.12.x` I think we should be able to resolve a bunch of the following envoy deprecation warnings easily with a straight replacement.

If any of these deprecations were introduced _after_ `v1.12.x` fixing these will require using the version sniffing logic recently added.

Deprecation Info:
https://www.envoyproxy.io/docs/envoy/latest/intro/deprecated 

Note this list is likely incomplete. For simplistic mesh scenarios these were just the first several that showed up in the envoy logs:

## Cluster Fields
* `envoy.deprecated_features:envoy.api.v2.Cluster.tls_context: true`
* `Using runtime overrides to continue using now fatal-by-default deprecated
option 'envoy.api.v2.Cluster.tls_context' from file cluster.proto.`
* `Using deprecated option 'envoy.api.v2.Cluster.hosts' from file cluster.proto.`

## HTTP
* `envoy.deprecated_features:envoy.config.trace.v2.ZipkinConfig.HTTP_JSON_V1: true`
* `envoy.deprecated_features:envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager.Tracing.operation_name: true`
* `Using runtime overrides to continue using now fatal-by-default deprecated
option
'envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager.Tracing.operation_name'
from file http_connection_manager.proto. `

## Listener Fields

* `Using deprecated option 'envoy.api.v2.listener.FilterChain.tls_context' from
file listener_components.proto.`
* `Using deprecated option 'envoy.api.v2.listener.Filter.config' from file
listener_components.proto.`

## Extension Names

* `Using deprecated extension name 'envoy.tcp_proxy' for
'envoy.filters.network.tcp_proxy'.`
* `Using deprecated extension name 'envoy.http_connection_manager' for
'envoy.filters.network.http_connection_manager'`
* `Using deprecated extension name 'envoy.router' for
'envoy.filters.http.router'.`

### PR DESCRIPTION
**Pull request by rboyer**
Note that this does NOT upgrade to xDS v3. That will come in a future PR.

Additionally:

- Ignored staticcheck warnings about how github.com/golang/protobuf is deprecated.
- Shuffled some agent/xds imports in advance of a later xDS v3 upgrade.
- Remove support for envoy `1.13.x` but don't add in `1.17.x` yet. We have to wait until the xDS v3 support is added in a follow-up PR.

Fixes #8425

### CODE REVIEW

File: agent/xds/listeners.go [hunk 1]
```diff
@@ -312,7 +311,7 @@ func (s *Server) makeIngressGatewayListeners(address string, cfgSnap *proxycfg.C
 			resources = append(resources, upstreamListener)
 		} else {
 			// If multiple upstreams share this port, make a special listener for the protocol.
-			listener := makeListener(listenerKey.Protocol, address, listenerKey.Port)
+			listener := makeListener(listenerKey.Protocol, address, listenerKey.Port, envoycore.TrafficDirection_OUTBOUND)
```
Comment (rboyer) [inline agent/xds/listeners.go:314]: This replaces the thing formerly controlled by the `ingress` boolean below.

File: agent/xds/listeners.go [hunk 2]
```diff
@@ -375,34 +379,16 @@ func makeListener(name, addr string, port int) *envoy.Listener {
 // any config generated by other systems will likely be in canonical protobuf
 // from rather than our slight variant in JSON/hcl.
 func makeListenerFromUserConfig(configJSON string) (*envoy.Listener, error) {
```
Comment (rboyer) [inline agent/xds/listeners.go:381]: It's typed configs everywhere now. This is a prerequisite for xDS v3, which is only typed.

File: agent/xds/listeners.go [hunk 3]
```diff
@@ -440,7 +429,7 @@ func (s *Server) injectHTTPFilterOnFilterChains(
 		)
 
 		for filterIdx, filter := range chain.Filters {
-			if filter.Name == wellknown.HTTPConnectionManager ||
```
Comment (rboyer) [inline agent/xds/listeners.go:443]: Stop using `wellknown` here because the value just changed.

File: command/connect/envoy/bootstrap_config.go [hunk 4]
```diff
@@ -550,14 +560,25 @@ func (c *BootstrapConfig) generateListenerConfig(args *BootstrapTplArgs, bindAdd
 		"connect_timeout": "5s",
 		"type": "STATIC",
 		"http_protocol_options": {},
-		"hosts": [
```
Comment (rboyer) [inline command/connect/envoy/bootstrap_config.go:553]: The `hosts` field was deprecated.

**Review by markan — APPROVED** [review]
I've read through the code changes in detail and this looks reasonable to me. 

@rboyer was kind enough to do a walkthrough, which greatly assisted with context

**Pull request merged**